linux/net/rina/pdu.c

```diff
@@ -287,3 +288,9 @@ int pdu_destroy(struct pdu * p)
         return 0;
 }
 EXPORT_SYMBOL(pdu_destroy);
+
+uint_t pdu_length(struct pdu * pdu) {
+	return buffer_length(pdu->buffer) +
+		serdes_pci_size(pci_type(pdu->pci));
+}
+EXPORT_SYMBOL(pdu_length);
```

Comment:
Yes, I have seen now this (
https://github.com/IRATI/stack/pull/760/files#diff-6b1399bf2f445bfa9cd35be0a403e2bfR1126).
I do not think this is ACKable because in the current form
```serdes_pci_size```
will not work if you change any configuration variable (seq num length, qos
leng
th and so on).
In the same way you crated ```serdes_pci_size``` in serdes.c, you should
cahnge it  to use the ```struct dt_cons``` for this. ```serdes_pci_size```
would become and static utility function within serdes.c and
```pdu_length``` in pdu.c should become something like ```ser_pdu_length```
in serdes.c

2015-11-10 13:19 GMT+01:00 Kewin Rausch <notifications@github.com>:

> In linux/net/rina/pdu.c
> <https://github.com/IRATI/stack/pull/760#discussion_r44399697>:
>
> > @@ -287,3 +288,9 @@ int pdu_destroy(struct pdu * p)
> >          return 0;
> >  }
> >  EXPORT_SYMBOL(pdu_destroy);
> > +
> > +uint_t pdu_length(struct pdu * pdu) {
> > +	return buffer_length(pdu->buffer) +
> > +		serdes_pci_size(pci_type(pdu->pci));
> > +}
> > +EXPORT_SYMBOL(pdu_length);
>
> Could be, I had some problem with this is serdes.c...
>
> I'm waiting for @sandervrijders <https://github.com/sandervrijders> ACK
> about this, because now serdes_pci_size is hardcoded with some values which
> are passed in the DIF configuration(addresses length and so on).
>
> The part which depends on the serializer is in fact handled by that
> serializer procedure, but this procedure has a pdu context (length of the
> pdu, not of it's data).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IRATI/stack/pull/760/files#r44399697>.
>



-- 
Leonardo Bergesio
Distributed Applications and Networks Area - DANA
i2CAT Foundation - www.i2cat.net
C/ Gran Capità 2-4, office 109, Nexus I building
08034 Barcelona, Catalonia (Spain)
T: +34 93 567 9926
F: +34 93 553 2520
W: http://dana.i2cat.net